tracing-aws/src/main/java/io.micronaut.tracing/AwsOpenTelemetryFactory.java

```diff
@@ -0,0 +1,106 @@
+import io.micronaut.context.annotation.Replaces;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.util.StringUtils;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.contrib.awsxray.AwsXrayIdGenerator;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.extension.aws.AwsXrayPropagator;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import jakarta.inject.Singleton;
+
+import java.util.Optional;
+
+/**
+ * Registers an OpenTelemetry bean.
+ *
+ * @author Nemanja Mikic
+ */
+@Factory
+@Replaces(factory = DefaultOpenTelemetryFactory.class)
+public class AwsOpenTelemetryFactory {
+
+    private static final String DEFAULT_OTEL_COLLECTOR_ENDPOINT = "http://localhost:4317";
+
+    /**
+     * The OpenTelemetry bean configured with AwsOpenTelemetryConfiguration.
+     * @param awsOpenTelemetryConfiguration the aws open-telemetry configuration
+     * @return the bean
+     */
+    @Singleton
+    @Primary
+    OpenTelemetry defaultOpenTelemetryWithConfig(@NonNull AwsOpenTelemetryConfiguration awsOpenTelemetryConfiguration) {
+
+        return OpenTelemetrySdk.builder()
+            // enables your downstream requests to include the X-Ray trace header
+            .setPropagators(
+                ContextPropagators.create(
+                    TextMapPropagator.composite(
+                        W3CTraceContextPropagator.getInstance(), AwsXrayPropagator.getInstance())))
+
+            // provides basic configuration of a TracerProvider which generates X-Ray compliant IDs
+            .setTracerProvider(
+                SdkTracerProvider.builder()
+                    .addSpanProcessor(
+                        BatchSpanProcessor.builder(
+                            OtlpGrpcSpanExporter.builder()
+                                .setEndpoint(
+                                    Optional.ofNullable(awsOpenTelemetryConfiguration.getOtlpGrpcEndpoint())
+                                    .filter(x -> !StringUtils.isEmpty(x))
+                                    .orElse(DEFAULT_OTEL_COLLECTOR_ENDPOINT))
+                                .build()
+                        ).build())
+                    .setIdGenerator(AwsXrayIdGenerator.getInstance())
+                    .build())
+            .build();
+    }
+
+    /**
+     * The OpenTelemetry bean with default values.
+     *
+     * @return the OpenTelemetry
+     */
+    @Singleton
+    OpenTelemetry defaultOpenTelemetry() {
+        return OpenTelemetrySdk.builder()
+            // enables your downstream requests to include the X-Ray trace header
+            .setPropagators(
```

Comment:
Duplicate `ContextPropagators.create`, probably it should be a bean built by the context.

Reply:
This whole module is surplus